executor/common_linux.h

```diff
@@ -2586,6 +2597,20 @@ static int do_sandbox_android(void)
 
 #define FS_IOC_SETFLAGS _IOW('f', 2, long)
 
+// The Android sandbox's seccomp policy blocks umount2.
+// In order for our process not to be trapped, we only call umount2 if needed.
+// On android, we should never have an extra mount.
+static int umount2_maybe(const char* target, int flags)
+{
+#if SYZ_EXECUTOR || SYZ_SANDBOX_ANDROID
```

Comment:
If I am reading this correctly, we stop doing umount in executor.
SYZ_SANDBOX_ANDROID (nor other sandbox modes) is not defined when building executor. We don't yet know what sandbox we will use (and can use several). Executor mode should support all sandboxes.
SYZ_SANDBOX_* and other SYZ_ defined are defined only when we create C reproducers, at that point we know sandbox and everything else.
If SYZ_EXECUTOR, we need some kind of dynamic dispatch. We have some global flags that contain sandbox, etc.

It would also be nice to not penalize all non-android reproducers with the additional no-op wrapper around umount2. Is it possible? If SYZ_SANDBOX_ANDROID we could strip _calls_ to umount (whole while loops).